docs/getting_started/go/hello_world_in_go/index.md

```diff
@@ -83,419 +85,767 @@ go get go.temporal.io/sdk
+	// Create the Temporal client
+	c, err := client.Dial(client.Options{})
+	if err != nil {
+		log.Fatalln("Unable to create Temporal client", err)
+	}
+	defer c.Close()
+
+	// Create the Temporal worker
+	w := worker.New(c, iplocate.TaskQueueName, worker.Options{})
+
+	// inject HTTP client into the Activities Struct
+	activities := &iplocate.IPActivities{
+		HTTPClient: http.DefaultClient,
+	}
+
+	// Register Workflow and Activities
+	w.RegisterWorkflow(iplocate.GetAddressFromIP)
+	w.RegisterActivity(activities)
+
+	// Start the Worker
+	err = w.Run(worker.InterruptCh())
+	if err != nil {
+		log.Fatalln("Unable to start Temporal worker", err)
+	}
 }
 
 ```
 <!--SNIPEND-->
 
-The `ComposeGreeting` Activity Definition also accepts a `Context` . This parameter is optional for Activity Definitions, but it's recommended because you may need it for other Go SDK features as your application evolves.
+The code imports the `iplocate` package, which includes your Workflow and Activity Definitions. It then defines a `main` function that creates and runs a Worker.
 
-The logic within the `ComposeGreeting` function creates the string and returns the greeting, along with `nil` for the error. In a more complex case, you can return an error from your Activity and then configure your Workflow to handle the error.
+You first create a client, and then you create a Worker that uses the client, along with the Task Queue it should listen on. By default, the client connects to the Temporal Cluster running at `localhost` on port `7233`, and connects to the `default` namespace. You can change this by setting values in `client.Options`.
 
-Your [Activity Definition](https://docs.temporal.io/activities#activity-definition) can accept input parameters just like Workflow Definitions. Review the [Activity parameters](https://docs.temporal.io/dev-guide/go/foundations#activity-parameters) section of the Temporal documentation for more details, as there are some limitations you'll want to be aware of when running more complex applications.
+Then you register your Workflow and Activities with the Worker. Since you defined your Activities as a struct, you use that instead of referencing your Activities directly. This is also where you inject the HTTP client so your Activities can access it.
 
-Like Workflow Definitions, if you have more than one parameter for an Activity, you should bundle the data into a struct rather than sending multiple input parameters. This will make future updates easier.
+In this case your Worker will run your Workflow and your two Activities, but there are cases where you could configure one Worker to run Activities, and another Worker to run the Workflows.
 
+Now you will use an `npm` script to start your Worker with Nodemon. Nodemon automatically reloads whenever it detects changes in your file, hence the command name `start.watch`. Be sure you have started the local Temporal Service and execute the following command to start your Worker:
```

Comment:
This paragraph looks like a holdover from the TS version.

Reply:
```suggestion
Now you'll start the Worker. Be sure you have started the local Temporal Service and execute the following command to start your Worker:
```